### DISCUSSION TITLE
Full Undo

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#7615] [Ideas]
At the moment, Scratch's undo is very limited. Indeed:
- if you select a different sprite in the sprite area, you cannot undo/redo changes you have done to the scripts of the other sprites (even if you select those sprites in the sprite area)
- you can only undo the last sprite/costume/sound removal. Any previous removal cannot be undone.

It would be really cool having a full undo, so that you can undo all the changes you have done to the scripts/sprites/costumes/sounds of the project.

**Comment by Joeclinton1** [comment]
Would be very time consuming to make; the kind of thing you'd get paid as software engineer to make. Basically every operation needs an undo operation, which needs to get pushed to the undo stack. Currently only certain, destructive operations have undo operations, so lots and lots of undo operations would need to be created. Additionally the functionality for the stack and the modification of the current system to use the stack instead of just replacing the currently active undo is not easy.

Scratch should do it themselves really.
